### ISSUE TITLE
Duplicate footnote references in Oral History transcripts result in illegal duplicate ids in HTML DOM

### ISSUE THREAD

**Issue by jrochkind** [sciencehistory/scihist_digicoll#1411]
It turns out duplicate footnote references are possible in our OH transcripts. See for instance [Smyth](https://digital.sciencehistory.org/works/4j03d097t) , which has duplicate references to footnotes 3 and 8. And conveniently happens to be the sample transcript we are using in our automated tests. 

When duplicate footnote references happen, our code ends up putting multiple elements into the HTML dom with the same `id` attribute -- eg `footnote8`. Because the `FootnoteReferenceComponent` ends up being rendered multiple times with the same `number ` argument. 

This is illegal HTML as well as an accessibility problem. 

<details>
<summary>See aXe checker warnings on Smyth:</summary>
Some labelled "critical" because of use in aria- attributes. 

```
      2) duplicate-id-aria: IDs used in ARIA and labels must be unique (critical)
           https://dequeuniversity.com/rules/axe/4.3/duplicate-id-aria?application=axeAPI
           The following 2 nodes violate this rule:

               Selector: #ohms_line_198 > .sr-only
               HTML: <span class="sr-only" id="footnote-text-3">3. See Beckman Center for the History of Chemistry oral history research file #0042 for a copy of undated typewritten autobiographical notes (pp.13).</span>
               Fix any of the following:
               - Document has multiple elements referenced with ARIA with the same id attribute: footnote-text-3

               Selector: #ohms_line_610 > .sr-only
               HTML: <span class="sr-only" id="footnote-text-8">
               Fix any of the following:
               - Document has multiple elements referenced with ARIA with the same id attribute: footnote-text-8

       3) duplicate-id: id attribute value must be unique (minor)
           https://dequeuniversity.com/rules/axe/4.3/duplicate-id?application=axeAPI
           The following 2 nodes violate this rule:

               Selector: #ohms_line_198 > a[name="footnote-reference3"]
               HTML: <a name="footnote-reference3" id="footnote-reference3"></a>
               Fix any of the following:
               - Document has multiple static elements with the same id attribute: footnote-reference3

               Selector: #ohms_line_610 > a[name="footnote-reference8"]
               HTML: <a name="footnote-reference8" id="footnote-reference8"></a>
               Fix any of the following:
               - Document has multiple static elements with the same id attribute: footnote-reference8
```

</details>

To fix this, I guess we need to keep track of footnote-references-seen, and on the second use, somehow either not use ID's in DOM at all, or use different IDs. 

The footnote itself is meant to have a link that takes you back to the footnote reference -- right now it kind of lets the browser decide which one, for a duplicate ID.  The obvious simple solution is it only takes you back to one of them, maybe the first one. Alternate solutions I'm not leaning toward, but just listing them for completeness: it could somehow list links to take you back to all references when multiple; it could use some kind of fancy javacript state to keep track of where you came from, and take you back to that one.  

I think the simple solution is probably sufficient at this point. 

Also, it's possible this fix would be easier if proceded by some refactoring/simplification of footnotes DOM/implementation, which I'll create another ticket for and link here.

**Comment by jrochkind** [comment]
So ordinarily, a footnote from the list at the bottom, it's number is a hyperlink back to the footnote reference in text. This is convenient, because you maybe click to see the footnote, now you can click again to get back to where you were. 

(Although the footnote tooltips mean you  might not have even clicked/scrolled, but let's say you had). 

If there are multiple references to eg footnote `5`, then... this is a problem, the link on footnote 5 can only take you "back" to one place. So...

1. It just only takes you back to the first reference, sorry if you came from second or third one you're on your own, and might get confused when it takes you back to first
2. It includes a list at the end of "back to first/second/third reference". 
3. It uses some kind of javascript to keep track of exactly where you came from and take you back to exactly that. (While this sounds fancy, and I guess is probably do-able... it's pretty complicated! Worth it for this weird edge case?)
4. Other?

Not sure what to do. @eddierubeiz , @apinkney0696 , thoughts?

**Comment by eddierubeiz** [comment]
I am leaning towards option 1, against option 2, and 3 sounds like a no to me.

There is also an option 4, which may be inferior to 1 but we should add it to the list. Don't sue me.

If there are multiple references to a footnote, repeat the entire citation in footnote and assign it its own number.

4: Change the content of the interview such that there is a one-to-one mapping between references and footnotes, with the understanding that in rare cases (and they are very rare) a footnote's entire content will be repeated.

Disadvantages:
 - you need to change all subsequent interview numbers in the interview
 - that's not how scholarly papers actually work
 - the footnote numbers themselves may be considered part of the interview and thus not subject to change just to make our lives easier
 - Validation code (at interview create / update time) *does* have to change. Subsequent interview uploads will need to enforce the new one-to-one mapping
 
 Advantages:
- Probably the most convenient to a casual user
- Arguably the most accessible solution
- This problem occurs very rarely, so not a lot of data to change
- No change to the footnote code.

**Comment by jrochkind** [comment]
4's an interesting idea, I don't know if it matters that it would be a divergence from the footnotes in the published/PDF transcript?

(1) is probably the easiest solution. 

(3) isn't totally off the table, but yeah I don't love it; another problem with it provides a solution only if you clicked on a reference to get to footnotes, if you scrolled down to read them instead, then what?

**Comment by eddierubeiz** [comment]
- The validation code in https://github.com/sciencehistory/scihist_digicoll/pull/746 can easily be repurposed to answer @jrochkind 's question of how rare repeated references are. (I believe the answer is `vanishingly`.)
- 1 is basically a more accessible version of the status quo, which I don't think is terrible, if indeed this problem almost never occurs.
- 3 is also likely to cost a lot of hours to maintain in the future.

**Comment by eddierubeiz** [comment]
Note also that `bundle exec rake scihist:check_ohms_xml_data` will run `OralHistoryContent::OhmsXmlValidator.new(model.ohms_xml_text).valid?` on every interview in the database.

**Comment by jrochkind** [comment]
Moving this back to "ready", I did some more preliminary refactoring in #1438, but didn't get to actually resolving the problem yet -- first we need to decide how.

**Comment by eddierubeiz** [comment]
OK. I think that is great.
